### DISCUSSION TITLE
v0.6.0-beta

### DISCUSSION THREAD

**Discussion by gbj** [leptos-rs/leptos#2193] [Announcements]
This is a breaking for our new server functions rewrite and Axum 0.7 support.

This should be a relatively feature-rich release, with limited breaking changes.

I'm interested in gathering feedback in the discussion associated with this release.

### Actix

- You can remove any explicit `.handle_server_fns()` call in your `main.rs`, as server functions are now handled in `.leptos_routes()`
 
### Axum
- This release supports Axum 0.7, so you'll need to migrate from Axum 0.6. The easiest way to do this is probably to consult [the diff on one of the examples](https://github.com/leptos-rs/leptos/pull/2158/files#diff-ceceec2ecd620c6891344b9407c391c6a9ca3b2d2f101881cd09ddd7f5a31981).
- You can remove any explicit `.handle_server_fns()` call in your `main.rs`, as server functions are now handled in `.leptos_routes()`

## Features

A rewritten server function system that is backwards-compatible, but reduces binary size and increases flexibility, specifically by allowing
  - automatic setup of server fn handlers with `.leptos_routes()` from the integrations
  - a variety of additional built-in encodings (rkyv, multipart forms/file uploads) in addition to the current set (GET URL, POST URL, CBOR, Rkyv) (#1868, #1989)
  - support for streaming responses from server functions (#1284)
  - ability to create custom user encodings for input and output by deriving `IntoReq`, `FromReq`, `IntoRes`, and/or `FromRes` traits
  - ability to mix and match encodings easily: This server function should be a JSON POST request and a ByteStream response, this one should be GET URL request and an rkyv response, etc.; any combination of the encodings above is supported
  - custom error types (#1657)
  - a `#[middleware]` macro to add per-server-function middleware from the Tower or Actix ecosystems (#1461)

Note: The additional included encodings (`serde_lite`, `rkyv`, multipart form data) are all enabled by additive features on the `server_fn` crate. If you want to use them you can just add that crate as a dependency and enable the required features.

**Example**: You can find a comprehensive example of these new features in the new [`server_fns_axum`](https://github.com/leptos-rs/leptos/blob/leptos_v0.6/examples/server_fns_axum/src/app.rs) example.

<hr /><em>This discussion was created from the release <a href='https://github.com/leptos-rs/leptos/releases/tag/v0.6.0-beta'>v0.6.0-beta</a>.</em>

**Comment by josephhajduk** [comment]
Just FYI,  I also needed to update my rust nightly
